pkg/kubelet/server.go

```diff
@@ -195,6 +196,16 @@ func (s *Server) hostnameHealthCheck(req *http.Request) error {
 	return nil
 }
 
+// Checks if kubelet's sync loop  that updates containers is working.
+func (s *Server) syncLoopHealthCheck(req *http.Request) error {
+	duration := s.host.ResyncInterval() * 2
```

Comment:
2 missed loops is not a lot when the resync interval is short; we need to track sync loop duration and make sure that this timeout is long enough to accommodate the longest sane operation that the sync loop might do.